### ISSUE TITLE
I can't fix this error (createCaussianFilter)

### ISSUE THREAD

**Issue by programmerpsk** [peterbraden/node-opencv#597]
when I use this module, I have a error like this (below)

OpenCV Error: Assertion failed (ksize.width > 0 && ksize.width % 2 == 1 && ksize.height > 0 && ksize.height % 2 == 1) in cv::createGaussianFilter, file C:\build\2_4_winpack-build-win64-vc14\opencv\modules\imgproc\src\smooth.cpp, line 816

How can I fix it?

**Comment by JonnyBGod** [comment]
I am getting a very similar error:

```
OpenCV Error: Assertion failed (0 <= anchor.x && anchor.x < ksize.width && 0 <= anchor.y && anchor.y < ksize.height) in init, file /tmp/opencv-20171112-53254-axs83r/opencv-3.3.1/modules/imgproc/src/filter.cpp, line 143
libc++abi.dylib: terminating with uncaught exception of type cv::Exception: /tmp/opencv-20171112-53254-axs83r/opencv-3.3.1/modules/imgproc/src/filter.cpp:143: error: (-215) 0 <= anchor.x && anchor.x < ksize.width && 0 <= anchor.y && anchor.y < ksize.height in function init
```

on macOs 10.13.2 (17C88)

**Comment by sivarajd** [comment]
I know it is an old thread. But I came across when I faced similar issue. So posting it here if it might help someone.

Something similar happened to me when using GaussianBlur with even numbers:

`blur = cv2.GaussianBlur(invert, (16, 16), cv2.BORDER_DEFAULT)`

Gave the error:

> cv2.error: OpenCV(4.10.0) /io/opencv/modules/imgproc/src/smooth.dispatch.cpp:293: error: (-215:Assertion failed) ksize.width > 0 && ksize.width % 2 == 1 && ksize.height > 0 && ksize.height % 2 == 1 in function 'createGaussianKernels'

When I changed the numbers in GaussianBlur from even to odd (like `blur = cv2.GaussianBlur(invert, (15, 15), cv2.BORDER_DEFAULT)` it got resolved.
